OpenSim/Common/Component.cpp

```diff
@@ -1090,6 +1094,26 @@ void Component::setNextSubcomponentInSystem(const Component& sub) const
 void Component::updateFromXMLNode(SimTK::Xml::Element& node, int versionNumber)
 {
     if (versionNumber < XMLDocument::getLatestVersion()) {
+        if (versionNumber < 30500) {
+            // In 3.3 and earlier, spaces in names were tolerated. Spaces are
+            // no longer acceptable in Component names.
+            if (node.hasAttribute("name")) {
+                auto name = node.getRequiredAttribute("name").getValue();
+                if (name.find_first_of("\n\t ") < std::string::npos) {
+                    std::cout << getConcreteClassName() << " name '" << name
+                        << "' contains whitespace. ";
+                    name.erase(
+                        std::remove_if(name.begin(), name.end(), ::isspace),
+                        name.end() );
+                    node.setAttributeValue("name", name);
+                    std::cout << "It is was renamed '" << name << "'." << std::endl;
+                }
+            }
+            else { // All Components should have a name. If none, assign one here.
```

Comment:
I'm guessing your unique-name-ifier code will catch duplicate names and append suffixes as necessary. Perhaps state that here in a comment (otherwise, it looks like something would end up breaking downstream if there are two or more nameless components of the same concrete class).